core/tests/protractor_utils/AdminPage.js

```diff
@@ -292,6 +334,190 @@ var AdminPage = function() {
     }
   };
 
+  this.getMiscTab = async function() {
+    await waitFor.elementToBeClickable(miscTabButton,
+      'Misc tab button not clickable');
+    await miscTabButton.click();
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.uploadTopicSimilarities =
+    async function(relativePathToSimilaritiesFile, isValidFile) {
+      var absPath = path.resolve(__dirname, relativePathToSimilaritiesFile);
+      await waitFor.visibilityOf(chooseSimilarityFileInput,
+        'Similarity upload form is not visible');
+      await chooseSimilarityFileInput.sendKeys(absPath);
+      await waitFor.elementToBeClickable(similarityFileUploadButton,
+        'Upload button taking too long to be clickable');
+      await similarityFileUploadButton.click();
+      if (isValidFile) {
+        var text = 'Topic similarities uploaded successfully.';
+        await waitFor.visibilityOf(statusMessage,
+          'Status message not visible');
+        await waitFor.textToBePresentInElement(statusMessage, text,
+          'Status message not visible');
+        expect(await statusMessage.getText()).toEqual(text);
+      } else {
+        var text = 'Server error: \'ascii\' codec can\'t encode characters' +
+          ' in position 1024-1025: ordinal not in range(128)';
+        await waitFor.visibilityOf(statusMessage,
+          'Status message not visible');
+        await waitFor.textToBePresentInElement(statusMessage,
+          text, 'Text not showing up in status message');
+        expect(await statusMessage.getText()).toEqual(text);
+      }
+    };
+
+  this.expectSimilaritiesToBeUploaded = async function() {
+    await waitFor.elementToBeClickable(similarityDownloadButton,
+      'Similarity upload failed – download similarities button not clickable.');
+  };
+
+  this.downloadSimilarityFile = async function() {
+    await waitFor.elementToBeClickable(similarityDownloadButton,
+      'Download similarity file button not clickable');
+    await similarityDownloadButton.click();
+  };
+
+  this.clearSearchIndex = async function() {
+    await waitFor.elementToBeClickable(searchIndexClearButton,
+      'Clear search index button not clickable');
+    await searchIndexClearButton.click();
+    await general.acceptAlert();
+  };
+
+  this.expectSearchIndexToBeCleared = async function() {
+    await waitFor.visibilityOf(statusMessage,
+      'Status message not showing up.');
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Index successfully cleared.');
+    expect(statusMessage.getText()).toEqual(
+      'Index successfully cleared.');
```

Comment:
Can you put this in `this.clearSearchIndex` instead? Typically the `expect...` functions check the long-term changes from an action, not the confirmation messages. Ditto for similar cases below

Reply:
done where applicable